components/ui/icons/src/main/res/drawable/mozac_ic_close.xml

```diff
@@ -3,10 +3,11 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-        android:width="24dp"
-        android:height="24dp"
-        android:viewportWidth="24.0"
-        android:viewportHeight="24.0">
-    <path android:fillColor="@color/mozac_ui_icons_fill"
-          android:pathData="M13.108 11.776l4.709-4.709c0.244-0.244 0.244-0.641 0-0.885s-0.64-0.244-0.884 0l-4.689 4.689h-0.488L7.067 6.183c-0.244-0.244-0.64-0.244-0.884 0s-0.244 0.641 0 0.885l4.687 4.686v0.491l-4.687 4.687c-0.244 0.244-0.244 0.641 0 0.885C6.305 17.939 6.465 18 6.625 18c0.16 0 0.32-0.061 0.442-0.183l4.687-4.687h0.491l4.687 4.687C17.055 17.939 17.215 18 17.375 18c0.16 0 0.32-0.061 0.442-0.183 0.244-0.244 0.244-0.641 0-0.885l-4.709-4.709v-0.447z"/>
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:fillColor="@color/mozac_ui_icons_fill"
+      android:pathData="M10.367,12.573L4.22,18.72C3.927,19.013 3.927,19.488 4.22,19.781C4.366,19.927 4.558,20 4.75,20C4.942,20 5.134,19.927 5.28,19.78L11.436,13.624H12.563L18.719,19.78C18.865,19.926 19.057,20 19.249,20C19.441,20 19.633,19.927 19.779,19.78C20.072,19.487 20.072,19.012 19.779,18.719L13.632,12.572L13.633,11.426L19.78,5.28C20.073,4.987 20.073,4.512 19.78,4.219C19.487,3.926 19.012,3.926 18.719,4.219L12.563,10.375H11.435L5.28,4.22C4.987,3.927 4.512,3.927 4.219,4.22C3.926,4.513 3.926,4.988 4.219,5.281L10.365,11.427L10.367,12.573Z" />
```

Comment:
```suggestion
      android:pathData="M10.367 12.573L4.22002 18.72C3.92702 19.013 3.92702 19.488 4.22002 19.781C4.36602 19.927 4.55802 20 4.75002 20C4.94202 20 5.13402 19.927 5.28002 19.78L11.436 13.624H12.563L18.719 19.78C18.865 19.926 19.057 20 19.249 20C19.441 20 19.633 19.927 19.779 19.78C20.072 19.487 20.072 19.012 19.779 18.719L13.632 12.572L13.633 11.426L19.78 5.28C20.073 4.987 20.073 4.512 19.78 4.219C19.487 3.926 19.012 3.926 18.719 4.219L12.563 10.375H11.435L5.28002 4.22C4.98702 3.927 4.51202 3.927 4.21902 4.22C3.92602 4.513 3.92602 4.988 4.21902 5.281L10.365 11.427L10.367 12.573Z" />
```
Can you use the unoptimized pathData here? There's no AS warning for performance issue with the original path data here since its under 600 characters. So, I don't really see any immediate benefit. As discussed on Slack, I would rather figure out a proper way to make our optimized SVG reproducible for everyone once we have unified the icons between Fenix and AC, and ensure we are using the latest icons from the Mobile Icons figma. 

The problem with optimizing the SVGs individually is that someone else can look at them and simply point out the path data does not match up with what they see from downloading the latest SVG and conclude the icon must be out of date.

Reply:
It's the path of a [file](https://www.figma.com/file/Kv7wWr8vH6AQrGhu9vmjdE/Mobile-Icons?node-id=221%3A1164) we get from figma, not optimized.

Reply:
I cannot reproduce this. Are you exporting the SVG via the Export on the right sidebar?
<img width="1200" alt="Screen Shot 2022-07-12 at 10 08 23 PM" src="https://user-images.githubusercontent.com/1190888/178635889-18502ad9-57b8-4fb1-82b7-be73d3f7c6db.png">

Reply:
Yes. And this svg looks like this:
<img width="794" alt="Screen Shot 2022-07-12 at 7 16 27 PM" src="https://user-images.githubusercontent.com/92760693/178636662-e0e3c220-ad67-4b71-8272-8f349e384d95.png">

which matches the change in PR

Reply:
Your screenshot matches the suggested changes I have posted in the first comment. Can you explain why you believe this is not the case?

Reply:
My suggestion here was to only copy and replace the existing path. There wasn't a need to add a new resource via AS since we are just updating the existing path. AS is only trimming the white and removing some of the extra floating point values, while yes this is an optimization, but it is not necessarily optimizing it the same as using SVGO for instance. I am not totally against landing it as is, but the problem will be that when we are checking to see if the existing icons are actually up to date with Figma, we would be comparing the optimized path vs the path you get from the exported SVG. If they are different, we would naturally go with the path you get from the exported SVG. The second step would be to find a common tool to optimize all the icons once we know they are up to date.

I would prefer if we just kept with our current methodology. We have already discussed the need to unify the icons between Fenix and AC, and to find a reproducible solution that can be applied to everything instead of having individual icons that are optimized, but unreproducible in terms of how to optimize by others. This goes beyond the scope of this issue and I am already the person working on the de-duplicating the icons between Fenix and AC. 

This solution could be the one that we have discussed which is that UX optimizes the SVGs that are exported so that it benefit both Android and iOS and serves as a single source of truth.

Reply:
`the problem will be that when we are checking to see if the existing icons are actually up to date with Figma`
The icons are already inconsistent, initially committed set of icons actually has [spaces removed.](https://github.com/pocmo/android-components/commit/765304aa66509eb0b7be982fc20d94112ee1e024#diff-3e6cfef369d20ad820e4f393ad48c4164123e4a305676742d6db4a41ea1cd743). The person who would be working on unifying icons state will have to go through every single icon anyway, so adding SVG through the resource manager doesn't make it worse.

So, we have two ways of adding SVG icons already. For the copy/past approach there is argument that it follows single source of truth, for the resource manager – optimization. I would like to go with the second one, if you don't have a strong opinion against it.

Btw, I have talked with the UX team, it's no really clear for them how the input to figma affects the output that we get, exporting icons. We should probably raise an issue to investigate that, and then approach the UX team with ideas how could we improve the flow.

Reply:
It's hard to infer the 'process' from https://github.com/pocmo/android-components/commit/765304aa66509eb0b7be982fc20d94112ee1e024#diff-3e6cfef369d20ad820e4f393ad48c4164123e4a305676742d6db4a41ea1cd743 since it was added in 2018. It is very likely that in 2018 that the SVG paths already have their spaces removed and optimized from what UX provided and no further optimization was done. Take for example https://searchfox.org/mozilla-central/source/browser/components/contextualidentity/content/cart.svg#7 which was added 2019 that we took and added in mozac_ic_cart.xml. We simply copied the path, but the path is already optimized.

The core problem is still the reproducibility as discussed and how to avoid confusion when telling whether or not an icon is up to date with the Mobile Icon figma. If the path is 1:1 from the exported icon in Figma, then we can absolutely tell if it's up to date or not, but any spacing or number changes might infer a change which might not be obvious even if you did a visual comparison. Since we don't know when icons are changed in the Figma, we just have to assume any difference means our icon is out of date.

My firm stance is to use the established process of copying and pasting 1:1 the path. I will carry out the followup with my existing work to unify the icons between Fenix and AC. This current process is already established and from speaking to Roger, what he would done as well.

Reply:
It was added in 2018 but icons like `mozac_ic_back` still have no spacing even after a year ago [update](https://github.com/mozilla-mobile/fenix/pull/19541/files#diff-5add202a098a9f4017b0ab359f825f9a77c6d1f65fd65e2e941f6c5239aaab38), which also has spacing removed. 

I assume icons with spacing removed were added either through AS resource manager (and got optimized) or got optimized by additional tools, while icons we export from figma have spacing. Am I correct? Do we have examples or icons exported from figma with commas instead of spaces?

Even more recent [uplifting of icons from fenix](https://github.com/gabrielluong/android-components/commit/c9703cc00baf5d55353cd076a441ac48017159ff#diff-dcaf77f44d39604cb9a1760811b5cf001cafc7d8996b3212ef9227d6636147d5) has spacing removed, not directly copied paths. So I have to doubt there is an established process that I should follow. Was there any discussion around it, any documented way that it should be done?

If not, I would like to follow optimized way. The smaller icon, while imported via the AS, goes from 1080 to 812 chars, which is slightly above the 800 warning mark.

Reply:
I am quite aware of Arturo's update to the proton icon for MR1 in 2021. He used https://github.com/alexjlockwood/avocado back then to optimize these icons. This was not fully fleshed out as the process nor documented. I have spoken to him at one point about creating a process that would optimize the SVGs but also reproducible. However, we are not at that point yet because of the following:
(1) We need to first remove the duplicates between Fenix and AC. This is the unifying step I am carrying out already. This is where we are currently are.
(2) Ensure the icons are up to date with the Figma icons. This will be carried via comparing the paths 1:1. Anything that is different have to be assumed is out of date as discussed before.
(3) Carryout a step that is either automated or manual or maybe not at all to optimize the SVGs. However, we need a way to know after this point if an icon is ever updated or a way to compare the old vs new.

I am not comfortable with amount of time spent in this discussion for what was suppose to fairly straight forward.

Reply:
I agree. I do believe we need a SVG optimization step, but I don't believe we are there yet. If we were to optimize our SVGs, the tooling or process should be consistent everywhere and done for all of the icons. For that reason, I am still most comfortable with checking in the 1:1 path. I have filed https://github.com/mozilla-mobile/android-components/issues/12502 as a tracking issue. As discussed above, the optimization step is around step 3 while we are still in the process of step 1 for that reasoning I don't believe we should land optimized SVGs until we know all the icons are using the up to date SVGs. This is well beyond the scope of this issue.

For what it is worth, having icons appear optimized doesn't necessarily mean we used an optimized tool. The SVG might've already had the whitespaces trimmed when they were designed and that was what was copy and pasted. The reason the optimization step should be a follow up is because we should align on the tool that is used - it is not necessarily gonna be AS, we might prefer SVGO or [avocado](https://github.com/alexjlockwood/avocado).